akka-persistence/src/main/scala/akka/persistence/state/javadsl/DurableStateUpdateStore.scala

```diff
@@ -20,5 +20,5 @@ trait DurableStateUpdateStore[A] extends DurableStateStore[A] {
    */
   def upsertObject(persistenceId: String, revision: Long, value: A, tag: String): CompletionStage[Done]
 
-  def deleteObject(persistenceId: String): CompletionStage[Done]
+  def deleteObject(persistenceId: String, revision: Long): CompletionStage[Done]
```

Comment:
ah, this is problematic for compatibility with existing implementations. However, this is more consistent with the `upsertObject` method. What do you think @johanandren ? Ok to break compatibility here because the method was not really used yet?

Reply:
A quick search shows a few impls already, better play it safe and provide an overload/deprecate?

Reply:
Should that be handled by calling both versions? Assuming the client should only implement one of them. I could see `NotImplementedException` being used by the client.
```
trait DurableStateUpdateStore[A] extends DurableStateStore[A] {

  /**
   * @param seqNr sequence number for optimistic locking. starts at 1.
   */
  def upsertObject(persistenceId: String, revision: Long, value: A, tag: String): Future[Done]

  @deprecated(message = "Use deleteObject with revision instead.", since = "2.6.20")
  def deleteObject(persistenceId: String): Future[Done]

  def deleteObject(persistenceId: String, revision: Long): Future[Done]

}
```
```
   protected def internalDelete(
      ctx: ActorContext[InternalProtocol],
      cmd: Any,
      state: Running.RunningState[S]): Running.RunningState[S] = {

    val newRunningState = state.nextRevision()
    val persistenceId = setup.persistenceId.id

    onDeleteInitiated(ctx, cmd)

    implicit val ec = ctx.executionContext

    val deleteObjectDeprecated = setup.durableStateStore.deleteObject(persistenceId).recover {
      case NotImplementedException => Done
    }

    val deleteObject = setup.durableStateStore.deleteObject(persistenceId, newRunningState.revision).recover {
      case NotImplementedException => Done
    }

    ctx.pipeToSelf[Done](Future.sequence(Seq(deleteObjectDeprecated, deleteObject)).map(_ => Done)) {
      case Success(_)     => InternalProtocol.DeleteSuccess
      case Failure(cause) => InternalProtocol.DeleteFailure(cause)
    }

    state.copy(state = setup.emptyState)
  }
```